### PR TITLE
Remove e2e-image-puller

### LINKED ISSUE #63355: e2e-image-puller doesn't work anymore.
The docker binary we use in COS and most images is dynamically linked.
```
$ file docker 
docker: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 2.6.32, BuildID[sha1]=45a7a62f39ddabd51f5eaad2f561df08fe048371, with debug_info, not stripped
```

It won't work if we directly mount it into busybox, which is what we are doing in e2e-image-puller https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/manifests/e2e-image-puller.manifest#L105.

Here is the e2e image puller log:
```
/bin/sh: docker: not found
/bin/sh: docker: not found
/bin/sh: docker: not found
06:33:34 pulling k8s.gcr.io/alpine-with-bash:1.0
06:33:34 pulling k8s.gcr.io/apparmor-loader:0.1
06:33:34 pulling k8s.gcr.io/busybox:1.24
06:33:34 pulling k8s.gcr.io/dnsutils:e2e
/bin/sh: docker: not found
/bin/sh: docker: not found
06:33:34 pulling k8s.gcr.io/e2e-net-amd64:1.0
06:33:34 pulling k8s.gcr.io/echoserver:1.6
/bin/sh: docker: not found
/bin/sh: docker: not found
06:33:34 pulling k8s.gcr.io/eptest:0.1
06:33:34 pulling k8s.gcr.io/fakegitserver:0.1
/bin/sh: docker: not found
06:33:34 pulling k8s.gcr.io/galera-install:0.1
06:33:34 pulling k8s.gcr.io/invalid-image:invalid-tag
/bin/sh: docker: not found
/bin/sh: docker: not found
06:33:34 pulling k8s.gcr.io/iperf:e2e
/bin/sh: docker: not found
/bin/sh: docker: not found
/bin/sh: docker: not found
/bin/sh: docker: not found
/bin/sh: docker: not found
06:33:34 pulling k8s.gcr.io/jessie-dnsutils:e2e
06:33:34 pulling k8s.gcr.io/k8s-dns-dnsmasq-amd64:1.14.5
06:33:34 pulling k8s.gcr.io/liveness:e2e
06:33:34 pulling k8s.gcr.io/logs-generator:v0.1.0
06:33:34 pulling k8s.gcr.io/mounttest:0.8
/bin/sh: docker: not found
/bin/sh: docker: not found
/bin/sh: docker: not found
/bin/sh: docker: not found
/bin/sh: docker: not found
/bin/sh: docker: not found
/bin/sh: docker: not found
...
```

I believe this has been broken for a long time. Given that the e2e test works fine till now without it, should we just remove it?

@yujuhong @kubernetes/sig-node-bugs @kubernetes/sig-testing-bugs

### PR DESCRIPTION
**Pull request by dims**
**What this PR does / why we need it**:

A long time ago, We added the image prepulling as a workaround due to
the overwhelming amount of flake caused by pulling during the tests.
This functionality has been broken for a while now when we switched to a
COS image where mounting `docker` binary into `busybox` stopped working.
So we just have dead code we should clean up.

Change-Id: I538171a5c1d9361eee7f9e0a99655b88b1721e3e

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #63355

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by dims** [comment]
cc @Random-Liu @yujuhong

**Review by cblecker — APPROVED** [review]
/approve

File: test/e2e/apps/network_partition.go [hunk 1]
```diff
@@ -106,12 +106,11 @@ var _ = SIGDescribe("Network Partition [Disruptive] [Slow]", func() {
...
 	var c clientset.Interface
 	var ns string
-	ignoreLabels := framework.ImagePullerLabels
 
 	BeforeEach(func() {
 		c = f.ClientSet
 		ns = f.Namespace.Name
-		_, err := framework.GetPodsInNamespace(c, ns, ignoreLabels)
+		_, err := framework.GetPodsInNamespace(c, ns, map[string]string{})
 		Expect(err).NotTo(HaveOccurred())
```
Comment (bowei) [inline test/e2e/apps/network_partition.go]: err is no longer being set here..

**Comment by bowei** [comment]
@dims -- I can review the code, but don't have enough context to approve :-)

**Comment by dims** [comment]
@bowei ack :)

cc @gmarek @jszczepkowski @vishh @mwielgus @MaciekPytel @jingax10 

folks, hoping one of you can approve this?

**Comment by spiffxp** [comment]
/lgtm
/retest
We're in code freeze now, I can be convinced this should be kicked out, but I don't want to kick it out just yet.

**Comment by spiffxp** [comment]
/assign @roberthbailey @mikedanese @zmerlynn
as top level cluster/ approvers, WDYT?

**Comment by roberthbailey** [comment]
I'm happy to approve the removal of the cluster bits. Adding a hold so that this doesn't merge until @spiffxp is ready.

/approve
/hold

**Comment by spiffxp** [comment]
/hold cancel
works for me

**Comment by dims** [comment]
/priority critical-urgent

oops looks like we need to bump the priority for this to merge

File: cluster/gce/manifests/e2e-image-puller.manifest [hunk 2]
```diff
@@ -1,117 +0,0 @@
...
-spec:
-  containers:
-  - name: image-puller
-    resources:
-      requests:
-        cpu: 50m
-      limits:
-        cpu: 50m
-    image: k8s.gcr.io/busybox:1.24
-    # TODO: Replace this with a go script that pulls in parallel?
```
Comment (BenTheElder) [inline cluster/gce/manifests/e2e-image-puller.manifest]: Did we ever build anything like this? I'd like to do this for some test clusters that may not have the fastest image pull and I suspect this is causing flakes for them. 
If there's at least a good way to dump the list of images somewhere that'd be really helpful. If not, maybe we could add something to dump the images for consumption by testing setups? I'd be happy to work on this.

**Pull request merged**